### PR TITLE
Add validation of epub:textref on body and seq elements

### PR DESCRIPTION
**Pull request by mattgarrish**
This PR makes the following changes:

- validates epub:textref when found on body or seq
- checks that a fragment has been specified in epub:textref and src attributes

### CODE REVIEW

File: src/main/java/com/adobe/epubcheck/overlay/OverlayHandler.java [hunk 1]
```diff
@@ -135,5 +140,20 @@ public void ignorableWhitespace(char[] chars, int arg1, int arg2)
...
+      report.message(MessageId.MED_014, EPUBLocation.create(path, parser.getLineNumber(), parser.getColumnNumber()));
+    }
+    
+    else {
+      String frag = PathUtil.getFragment(ref.trim());
+      if (Strings.isNullOrEmpty(frag)) {
+        // empty fragid # not allowed
+        report.message(MessageId.MED_015, EPUBLocation.create(path, parser.getLineNumber(), parser.getColumnNumber()));
+      }
+    }
```
Comment (rdeltour) [inline src/main/java/com/adobe/epubcheck/overlay/OverlayHandler.java:157]: If we want to only create one message for this (see the comment below), we can combine the logic with something like:

```java
    if (ref.indexOf("#") == -1
         || Strings.isNullOrEmpty(PathUtil.getFragment(ref.trim()))) {
        // must include a non-empty fragment identifier
        report.message(MessageId.MED_014, EPUBLocation.create(path, parser.getLineNumber(), parser.getColumnNumber()));
      }
```

File: src/main/resources/com/adobe/epubcheck/messages/MessageBundle.properties [hunk 2]
```diff
@@ -131,6 +131,8 @@ MED_004=Image file header may be corrupted.
 MED_005=Media Overlay audio reference %1$s to non-standard audio type %2$s found.
 MED_006=Some browsers do not support rendering SVG images which use a filename in the xlink:href property.
 MED_007=Foreign resources can only be referenced from "source" elements with an explicit "type" attribute; found resource "%1$s" of foreign type "%2$s".
+MED_014=A fragment identifier is required.
+MED_015=Empty fragment identifiers are not allowed.
```
Comment (rdeltour) [inline src/main/resources/com/adobe/epubcheck/messages/MessageBundle.properties:135]: I'm not sure this warrants two distinct messages, maybe we can combine these into only `MED-014`? ("A (non-empty) fragment identifier is required")
WDYT?

File: src/test/resources/epub3/mediaoverlays-publication.feature [hunk 3]
```diff
@@ -22,7 +22,14 @@ Feature: EPUB 3 ▸ Media Overlays ▸ Full Publication Checks
 
   ## 2. Media Overlay Document Definition
 
-  ### The audio Element
+  Scenario: Report invalid fragment identifiers on body, seq, and text elements
+    When checking EPUB 'mediaoverlays-invalid-fragid-error'
+    Then error MED-014 is reported
+    And error MED-015 is reported
+    And error RSC-012 is reported
+    And no other errors or warnings are reported
```
Comment (rdeltour) [inline src/test/resources/epub3/mediaoverlays-publication.feature:30]: I suggest the test is split in two:
- one test for missing or empty fragment id (`mediaoverlays-text-src-fragment-missing-error`), to test `MED-014/MED-015`
- one test for a fragment not found in the content doc (`mediaoverlays-text-src-fragment-not-found-in-content-error`), to test `RSC-012`

**Comment by mattgarrish** [comment]
A few headaches trying to hand edit some conflicts in the github editor, but the requested changes should now be resolved.

**Review by rdeltour — APPROVED** [review]

**Pull request merged**